src/api/string_methods.cc

```diff
@@ -65,8 +65,10 @@ void NewTaintedString(const FunctionCallbackInfo<Value>& args) {
     auto parameterName = args[2];
     auto type = args[3];
 
-    // if string length == 1 then make a new one in order to avoid chache issues.
+    // if string length < 10 then make a new one in order to avoid cache issues.
     if (v8::Local<v8::String>::Cast(args[1])->Length() == 1) {
+        parameterValue = tainted::NewExternalString(isolate, args[1]);
+    } else if (v8::Local<v8::String>::Cast(args[1])->Length() < 10) {
```

Comment:
```suggestion 
    // if string length < 10 then make a new one in order to avoid cache issues.
    int len =  v8::Local<v8::String>::Cast(args[1])->Length();
    if (len == 1) {
        parameterValue = tainted::NewExternalString(isolate, args[1]);
    } else if (len < 10) {
```